### PR TITLE
Initial stub for demo 2

### PR DESCRIPTION
**Pull request by nitro-marky**
This is a basic beginning point for demo 2 when clicked on. The copy will need updating as will the interface.

There is also a basic Cypress test.

The current component is called `AccessDemo `  and can be renamed, moved as seen fit.

Screenshot 2022-08-30 at 14 00 53

Screenshot 2022-08-30 at 13 59 59

### CODE REVIEW

**Review by mattdean-digicatapult — COMMENTED** [review]
Couple of comments on some things we might want to change.

File: src/fixtures/demos.js [hunk 1]
```diff
@@ -35,10 +35,13 @@ export const demos = [
   },
   {
     path: 'demo2',
+    binaryName: 'out-of-bounds-access',
     title: 'Do you think your files are safe?',
```
[the remark this answers is not in the document]
Comment (ls1874) [reply src/fixtures/demos.js:40]: @mattdean-digicatapult @nitro-marky Maybe something like `Is your password what you think it is? `or `Is your login really secure?` Hard to think of something informative but short.

File: src/components/Component.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import React from 'react'
+import Header from './Header'
+
+export default function Component(props) {
```
Comment (n3op2) [inline src/components/Component.js:4]: Same, maybe change name to something a bit more scenario related? @mattdean-digicatapult

File: src/components/Header.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ export default function Header({ logo, showClose, title, ...props }) {
   const { update } = React.useContext(Context)
 
   return (
-    <HeaderStyle backgroundColor={props.color || 'none'}>
+    <HeaderStyle id={'header'} backgroundColor={props.color || 'none'}>
```
Comment (n3op2) [inline src/components/Header.js:27]: if this is for cypress testing the good idea is to prefix with soething like `cy-` https://docs.cypress.io/guides/references/best-practices#Selecting-Elements

Comment (nitro-marky) [reply src/components/Header.js]: Done

File: src/components/Component.js (hunk 2 above)
Comment (mattdean-digicatapult) [reply src/components/Component.js:4]: let's call it `AccessDemo`

Comment (nitro-marky) [reply src/components/Component.js]: Done

File: cypress/integration/demo-two.spec.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+describe('Demo Two', () => {
+  beforeEach(() => {
+    cy.visit('/demo2')
+  })
+
+  it('Renders', () => {
+    cy.get('[data-cy=header]').should('be.visible')
```
Comment (n3op2) [inline cypress/integration/demo-two.spec.js:7]: nice!!!

**Review by n3op2 — APPROVED** [review]

**Pull request merged**